orcid_hub/models.py

```diff
@@ -644,7 +644,8 @@ def is_superuser(self, value):  # noqa: D401
         """Sets user as a HUB admin."""
         if value:
             self.roles |= Role.SUPERUSER
-        else:
+        # Only do the bitwise XOR in case if superuser is present in roles,otherwise superuser gets set in user edit.
+        elif bool(self.roles & Role.SUPERUSER):
```

Comment:
there was indeed an error in the logic, but now it is even more convoluted. So basically we need to set on or off the flag corresponding the superuser role. I had had to be:

```
ON: self.roles != Role.SUPERUSER
OFF: self.roles &= ~Role.SUPERUSER
```
My bad! Can you fix it. There is no need to use XOR 

```
>>> u.roles |= models.Role.SUPERUSER

>>> u.roles
<Role.RESEARCHER|SUPERUSER: 5>

>>> u.roles &= ~models.Role.SUPERUSER

>>> u.roles
<Role.RESEARCHER: 4>
```

Reply:
Thanks for suggestion, Done changes ( Github: https://github.com/Royal-Society-of-New-Zealand/NZ-ORCID-Hub/pull/456 )